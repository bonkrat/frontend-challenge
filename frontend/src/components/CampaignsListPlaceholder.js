import React from "react";
import { classnames } from "tailwindcss-classnames";

const root = classnames("flex", "flex-col");
const row = classnames("flex", "h-64", "flex-grow", "px-2", "mb-2");
const heroImage = classnames("w-1/6", "bg-gray-100");
const heroDescription = classnames("bg-gray-100", "mx-4", "w-5/6");
const image = classnames("w-1/2", "bg-gray-100");
const description = classnames("bg-gray-100", "mx-6", "w-1/2");

/**
 * Placeholder for campaigns while they load.
 */
const CampaignListPlaceholder = () => (
  <div className={root}>
    <div className={row}>
      <div className={heroImage}></div>
      <div className={heroDescription}></div>
    </div>

    <div className={row}>
      <div className={image}></div>
      <div className={description}></div>
    </div>

    <div className={row}>
      <div className={image}></div>
      <div className={description}></div>
    </div>
  </div>
);

export default CampaignListPlaceholder;
