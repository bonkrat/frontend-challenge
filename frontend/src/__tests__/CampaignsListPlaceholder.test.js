import React from "react";
import { shallow } from "enzyme";
import CampaignsListPlaceholder from "../components/CampaignsListPlaceholder";
import toJson from "enzyme-to-json";

it("renders correctly", () => {
  expect(toJson(shallow(<CampaignsListPlaceholder />))).toMatchSnapshot();
});
