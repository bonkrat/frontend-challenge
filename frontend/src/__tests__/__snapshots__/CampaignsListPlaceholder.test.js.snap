// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-col"
>
  <div
    className="flex h-64 flex-grow px-2 mb-2"
  >
    <div
      className="w-1/6 bg-gray-100"
    />
    <div
      className="bg-gray-100 mx-4 w-5/6"
    />
  </div>
  <div
    className="flex h-64 flex-grow px-2 mb-2"
  >
    <div
      className="w-1/2 bg-gray-100"
    />
    <div
      className="bg-gray-100 mx-6 w-1/2"
    />
  </div>
  <div
    className="flex h-64 flex-grow px-2 mb-2"
  >
    <div
      className="w-1/2 bg-gray-100"
    />
    <div
      className="bg-gray-100 mx-6 w-1/2"
    />
  </div>
</div>
`;
